ibm/resource_ibm_cos_bucket.go

```diff
@@ -673,6 +804,17 @@ func resourceIBMCOSRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	retentionInput := &s3.GetBucketProtectionConfigurationInput{
+		Bucket: aws.String(bucketName),
+	}
+	retentionptr, err := s3Client.GetBucketProtectionConfiguration(retentionInput)
+	if err != nil {
```

Comment:
Do we need to add if (err != nil && !strings.Contains(err.Error(), "NoSuchLifecycleConfiguration: The lifecycle configuration does not exist")) && (err != nil && bucketPtr != nil && bucketPtr.Firewall != nil && !strings.Contains(err.Error(), "AccessDenied: Access Denied")) {
similar to expiry and other rule....may be can you try creating a cos bucket along with allowd_ip and retention policy and test it